### ISSUE TITLE
derivable_impls false positive on impl with doc comment

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#7654]
```rust
pub struct Color {
    pub r: u8,
    pub g: u8,
    pub b: u8,
}

/// `#000000`
impl Default for Color {
    fn default() -> Self {
        Color { r: 0, g: 0, b: 0 }
    }
}
```

```console
$ cargo clippy

warning: this `impl` can be derived
  --> src/main.rs:8:1
   |
8  | / impl Default for Color {
9  | |     fn default() -> Self {
10 | |         Color { r: 0, g: 0, b: 0 }
11 | |     }
12 | | }
   | |_^
   |
   = note: `#[warn(clippy::derivable_impls)]` on by default
   = help: try annotating `Color` with `#[derive(Default)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls
```

Clippy's suggestion is not appropriate in this case because the impl carries documentation, which is not substitutable by a derived impl.

![Screenshot from 2021-09-09 19-14-34](https://user-images.githubusercontent.com/1940490/132788365-b0d35352-be8b-46c7-a4ef-9e52f00d484e.png)

@HKalbasi @camsteffen

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.57.0-nightly (497ee321a 2021-09-09)
binary: rustc
commit-hash: 497ee321af3b8496eaccd7af7b437f18bab81abf
commit-date: 2021-09-09
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```
